touchforms/backend/persistence.py

```diff
@@ -24,32 +26,114 @@ def restore(sess_id, factory, override_state=None):
         state.update(override_state)
     return factory(**state)
 
-# TODO integrate with real caching framework (django ideally)
-def cache_set(key, value, timeout):
-    with open(cache_path(key), 'w') as f:
-        f.write(json.dumps(value).encode('utf8'))
 
-# TODO integrate with real caching framework (django ideally)
+def cache_set(key, value):
+    if key is None:
+        raise KeyError
+    postgres_insert(key, value)
+
+
 def cache_get(key):
+    if key is None:
+        raise KeyError
+    try:
+        return postgres_lookup(key)
+    except KeyError:
+        return cache_get_old(key)
```

Comment:
if we keep fylesystem support you should probably rename this to `cache_get_flie`